main.go

```diff
@@ -129,116 +143,132 @@ func main() {
 			os.Exit(1)
 		}
-	}
+		prometheus.MustRegister(exporter)
+
+		// TODO(@sysadmind): Remove this when we have a better way to get the cluster name to down stream collectors.
+		// cluster info retriever
+		clusterInfoRetriever := clusterinfo.New(logger, httpClient, esURL, *esClusterInfoInterval)
+
+		prometheus.MustRegister(collector.NewClusterHealth(logger, httpClient, esURL))
+		prometheus.MustRegister(collector.NewNodes(logger, httpClient, esURL, *esAllNodes, *esNode))
+
+		if *esExportIndices || *esExportShards {
+			sC := collector.NewShards(logger, httpClient, esURL)
+			prometheus.MustRegister(sC)
+			iC := collector.NewIndices(logger, httpClient, esURL, *esExportShards, *esExportIndexAliases)
+			prometheus.MustRegister(iC)
+			if registerErr := clusterInfoRetriever.RegisterConsumer(iC); registerErr != nil {
+				logger.Error("failed to register indices collector in cluster info")
+				os.Exit(1)
+			}
+			if registerErr := clusterInfoRetriever.RegisterConsumer(sC); registerErr != nil {
+				logger.Error("failed to register shards collector in cluster info")
+				os.Exit(1)
+			}
+		}
 
-	if *esExportIndicesSettings {
-		prometheus.MustRegister(collector.NewIndicesSettings(logger, httpClient, esURL))
-	}
+		if *esExportIndicesSettings {
+			prometheus.MustRegister(collector.NewIndicesSettings(logger, httpClient, esURL))
+		}
 
-	if *esExportIndicesMappings {
-		prometheus.MustRegister(collector.NewIndicesMappings(logger, httpClient, esURL))
-	}
+		if *esExportIndicesMappings {
+			prometheus.MustRegister(collector.NewIndicesMappings(logger, httpClient, esURL))
+		}
 
-	// Create a context that is cancelled on SIGKILL or SIGINT.
-	ctx, cancel := signal.NotifyContext(context.Background(), os.Interrupt, os.Kill)
-	defer cancel()
+		// start the cluster info retriever
+		switch runErr := clusterInfoRetriever.Run(ctx); runErr {
+		case nil:
+			logger.Info("started cluster info retriever", "interval", (*esClusterInfoInterval).String())
+		case clusterinfo.ErrInitialCallTimeout:
+			logger.Info("initial cluster info call timed out")
+		default:
+			var err error
+			logger.Error("failed to run cluster info retriever", "err", err)
+			os.Exit(1)
+		}
 
-	// start the cluster info retriever
-	switch runErr := clusterInfoRetriever.Run(ctx); runErr {
-	case nil:
-		logger.Info("started cluster info retriever", "interval", (*esClusterInfoInterval).String())
-	case clusterinfo.ErrInitialCallTimeout:
-		logger.Info("initial cluster info call timed out")
-	default:
-		logger.Error("failed to run cluster info retriever", "err", err)
-		os.Exit(1)
+		// register cluster info retriever as prometheus collector
+		prometheus.MustRegister(clusterInfoRetriever)
 	}
 
-	// register cluster info retriever as prometheus collector
-	prometheus.MustRegister(clusterInfoRetriever)
+	http.HandleFunc(*metricsPath, func(w http.ResponseWriter, r *http.Request) {
+		// If query has target param treat like probe endpoint
+		if r.URL.Query().Has("target") {
```

Comment:
I'm not sure that I see the value in having the ability to swap to multi-target support here. The /probe endpoint could do things that the /metrics endpoint would not which would be unexpected for users. I don't know of any exporters that handle the target parameter like this so I don't know that it's a good idea to do so here.